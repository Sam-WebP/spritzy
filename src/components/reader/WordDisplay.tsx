import { WordParts, ReaderSettings } from '@/types';

interface WordDisplayProps {
  currentWord: WordParts;
  settings: ReaderSettings;
}

export default function WordDisplay({ currentWord, settings }: WordDisplayProps) {
  return (
    <div 
      className="h-20 flex items-center justify-center border-2 border-gray-300 rounded-lg mb-4"
      style={{ backgroundColor: settings.theme.wordBackground }}
      aria-live="assertive"
      aria-atomic="true"
    >
      <div 
        className={`${settings.font.className} flex items-baseline relative`}
        style={{ 
          fontSize: `${settings.fontSize}px`, 
          letterSpacing: `${settings.letterSpacing}px` 
        }}
      >
        <div style={{ color: settings.theme.text }}>{currentWord.before}</div>
        {settings.showFocusLetter ? (
          <div 
            style={{ 
              color: settings.theme.highlightText,
              borderBottom: settings.showFocusBorder ? `2px solid ${settings.theme.highlightBorder}` : 'none',
              display: 'inline-block', // Fix extra padding issue
              lineHeight: '1',
            }}
          >
            {currentWord.pivot}
          </div>
        ) : (
          <div style={{ color: settings.theme.text }}>{currentWord.pivot}</div>
        )}
        <div style={{ color: settings.theme.text }}>{currentWord.after}</div>
      </div>
    </div>
  );
}