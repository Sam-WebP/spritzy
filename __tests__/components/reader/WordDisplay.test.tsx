import { render, screen } from '@testing-library/react';
import WordDisplay from '@/components/reader/WordDisplay';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import readerReducer from '@/redux/slices/readerSlice';
import settingsReducer from '@/redux/slices/settingsSlice';

const createTestStore = (initialState = {}) => {
    return configureStore({
        reducer: {
            reader: readerReducer,
            settings: settingsReducer
        },
        preloadedState: initialState
    });
};

describe('WordDisplay', () => {
    it('renders word parts correctly', () => {
        const store = createTestStore({
            reader: {
                currentWord: { before: 'he', pivot: 'l', after: 'lo' }
            },
            settings: {
                font: { className: 'font-mono' },
                fontSize: 24,
                letterSpacing: 0,
                showFocusLetter: true,
                showFocusBorder: true
            }
        });

        render(
            <Provider store={store}>
                <WordDisplay />
            </Provider>
        );

        expect(screen.getByText('he')).toBeInTheDocument();
        expect(screen.getByText('l')).toBeInTheDocument();
        expect(screen.getByText('lo')).toBeInTheDocument();
    });

    it('applies font settings correctly', () => {
        const store = createTestStore({
            reader: {
                currentWord: { before: 'h', pivot: 'e', after: 'llo' }
            },
            settings: {
                font: { className: 'font-sans' },
                fontSize: 32,
                letterSpacing: 1,
                showFocusLetter: true,
                showFocusBorder: true
            }
        });

        const { container } = render(
            <Provider store={store}>
                <WordDisplay />
            </Provider>
        );

        const wrapper = container.firstChild?.firstChild?.firstChild;
        expect(wrapper).toHaveClass('font-sans');
        expect(wrapper).toHaveStyle({ fontSize: '32px', letterSpacing: '1px' });
    });

    it('highlights pivot letter when showFocusLetter is true', () => {
        const store = createTestStore({
            reader: {
                currentWord: { before: 'h', pivot: 'e', after: 'llo' }
            },
            settings: {
                font: { className: 'font-mono' },
                fontSize: 24,
                letterSpacing: 0,
                showFocusLetter: true,
                showFocusBorder: false
            }
        });

        render(
            <Provider store={store}>
                <WordDisplay />
            </Provider>
        );

        const pivot = screen.getByText('e');
        expect(pivot).toHaveClass('text-primary');
    });

    it('shows focus border when showFocusBorder is true', () => {
        const store = createTestStore({
            reader: {
                currentWord: { before: 'h', pivot: 'e', after: 'llo' }
            },
            settings: {
                font: { className: 'font-mono' },
                fontSize: 24,
                letterSpacing: 0,
                showFocusLetter: false,
                showFocusBorder: true
            }
        });

        render(
            <Provider store={store}>
                <WordDisplay />
            </Provider>
        );

        const pivot = screen.getByText('e');
        expect(pivot).toHaveClass('border-b-2');
    });
});
