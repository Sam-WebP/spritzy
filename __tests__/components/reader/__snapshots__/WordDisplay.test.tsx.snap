// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WordDisplay > handles empty word parts gracefully 1`] = `
<div
  class="rounded-lg border text-card-foreground shadow-sm h-20 bg-background"
>
  <div
    class="h-full flex items-center justify-center p-2"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="w-full"
    />
  </div>
</div>
`;

exports[`WordDisplay > handles null currentWord gracefully 1`] = `
<div
  class="rounded-lg border text-card-foreground shadow-sm h-20 bg-background"
>
  <div
    class="h-full flex items-center justify-center p-2"
  />
</div>
`;
